### PR TITLE
Feature/issue 329 & 331 bookmarked topics, recommended topics, and options

### PR DESCRIPTION
**Pull request by bensinjin**
This PR includes #329 & #331. It's worth checking this branch out and running it on a device in order to get a better feel visually as to the affects of the changes.

### CODE REVIEW

**Comment by bensinjin** [comment]
@rasmus-storjohann-PG sooner rather than later for a review on this one, naturally it touches quite a  few files.

File: src/components/tasks/task_detail_component.tsx [hunk 1]
```diff
@@ -28,6 +29,7 @@ type Props = TaskDetailProps & TaskDetailActions;
 export const TaskDetailComponent: React.StatelessComponent<Props> = (props: Props): JSX.Element => {
     const task = props.task;
     const onServicesButtonPress = goToRouteWithParameter(Routes.Services, task.id, props.history);
+    const bookmarkProps = { taskId: props.task.id, addBookmark: props.addToSavedList, removeBookmark: props.removeFromSavedList };
```
Comment (rasmus-storjohann-PG) [inline src/components/tasks/task_detail_component.tsx]: Try to keep to the line length limit, this line can easily be broken up for improved legibility

File: src/components/tasks/task_list_item_component.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ type Props = TaskListItemProps & TaskListItemActions;
 
 export const TaskListItemComponent: React.StatelessComponent<Props> = (props: Props): JSX.Element => {
     const taskDescription = stripMarkdown(props.task.description);
+    const bookmarkProps = { taskId: props.task.id, addBookmark: props.addToSavedList, removeBookmark: props.removeFromSavedList };
```
Comment (rasmus-storjohann-PG) [inline src/components/tasks/task_list_item_component.tsx]: Another long line

**Comment by rasmus-storjohann-PG** [comment]
QA issue: Open a topic with lots of related topics, and scroll down the list of related topics. Click on a related topic. Expected is that I will go to the top of the screen for the newly selected topic. Observed is that I'm scrolled down on the new page.

File: src/selectors/tasks/get_newly_recommended_tasks.ts [hunk 3]
```diff
@@ -5,9 +5,11 @@ import * as R from 'ramda';
...
-            getRecommendedTasks(oldAnswers, tasks, savedTaskIds),
-        )
-    );
+    (oldAnswers: AnswersMap, newAnswers: AnswersMap, tasks: TaskMap, savedTaskIds: ReadonlyArray<Id>): ReadonlyArray<Task> => {
+        const newTasksFromAnswers = R.difference(
+            getRecommendedTasks(newAnswers, tasks),
+            getRecommendedTasks(oldAnswers, tasks),
+        );
+        const taskIsNotSaved = (task: Task): boolean => R.not(R.contains(task.id, savedTaskIds));
+        return R.filter(taskIsNotSaved, newTasksFromAnswers);
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/selectors/tasks/get_newly_recommended_tasks.ts]: I've renamed this to better indicate it's "recommended unsaved" tasks. This used to be what we previously called just "recommended" which was actually somewhat misleading. Thanks for the reject tip, always forget that one...

File: src/components/home/home_component.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                full
+                onPress={goToRouteWithoutParameter(Routes.Questionnaire, props.history)}
+                style={[applicationStyles.whiteButton, applicationStyles.boxShadowBelow]}
+            >
+                <Text style={textStyles.whiteButton}>
+                    <Trans>Start</Trans>
+                </Text>
+            </Button>
+        </View>
+    );
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/home/home_component.tsx]: Fixed

File: src/components/recommended_topics/recommended_topics_connected_component.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import { Store } from '../../stores';
+import { RecommendedTopicsComponent, RecommendedTopicsProps } from './recommended_topics_component';
+import { selectRecommendedTasks } from '../../selectors/tasks/select_recommended_tasks';
+import { Id, AddToSavedListAction, addToSavedList, RemoveFromSavedListAction, removeFromSavedList } from '../../stores/tasks';
+import { pickSavedTaskIds } from '../../selectors/tasks/pick_saved_task_ids';
+import { getIdsOfChosenAnswers } from '../../selectors/questionnaire/get_ids_of_chosen_answers';
+import { pickAnswers } from '../../selectors/questionnaire/pick_answers';
+
+const mapStateToProps = (store: Store): RecommendedTopicsProps => ({
+    chosenAnswers: getIdsOfChosenAnswers(pickAnswers(store)),
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/recommended_topics/recommended_topics_connected_component.tsx]: Fixed

File: src/components/tasks/bookmark_button_component.tsx [hunk 6]
```diff
@@ -17,21 +15,37 @@ export interface BookmarkButtonActions {
...
+    buildBookmarkForDetail((): AddToSavedListAction => props.addBookmark(props.taskId), getAddBookmarkIcon())
+);
+
+export const DetailRemoveBookmarkComponent: React.StatelessComponent<Props> = (props: Props): JSX.Element => (
+    buildBookmarkForDetail((): RemoveFromSavedListAction => props.removeBookmark(props.taskId), getRemoveBookmarkIcon())
+);
+
+const getAddBookmarkIcon = (): JSX.Element => (
+    <Icon type='FontAwesome' style={{ color: colors.topaz }} name={'bookmark-o'} />
+);
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/tasks/bookmark_button_component.tsx]: Fixed

File: src/components/tasks/task_detail_component.tsx (hunk 1 above)
Comment (bensinjin) [reply src/components/tasks/task_detail_component.tsx]: Fixed

File: src/components/tasks/task_list_item_component.tsx (hunk 2 above)
Comment (bensinjin) [reply src/components/tasks/task_list_item_component.tsx]: Fixed

**Comment by bensinjin** [comment]
Good point about long lists and navigating to new screens. This will need to be addressed app wide and I sort of think we may be able to get the router to do the work (ie. when we go to a new route scroll to the top of the page), if that's not the case we can manually scroll as I've done in other places but it's a bit of pain. Can you open a story for this?

**Comment by bensinjin** [comment]
On closer inspection of the scrolling issue I'm noticing it seems to only occur when jumping between what I call "child" pages, pages with routes like <some-thing>/<some-id>. When going from say /bookmarked-topics to topic/A topic Id the page will scroll to the top.

File: src/selectors/tasks/get_newly_recommended_tasks.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (rasmus-storjohann-PG) [reply src/selectors/tasks/get_newly_recommended_tasks.ts]: You're right, this is actually cleaner. But I think we can make it even better. I suggest that we keep the `getNewlyRecommended()` name and remove all the logic related to saved state from it, and then in `selectNewlyRecommendedUnsavedTasks()` we add a call to remove all saved tasks there. Then we have `getNewlyRecommended()` and the new `rejectSavedTasks()` as nice clean, single purpose reusable functions and the top level selector pulling everything together for consumption by the UX.

[the remark this answers is not in the document]
Comment (rasmus-storjohann-PG) [reply src/selectors/tasks/get_newly_recommended_tasks.ts]: @DarwishMenna this discussion may be of interest to you

[the remark this answers is not in the document]
Comment (bensinjin) [reply src/selectors/tasks/get_newly_recommended_tasks.ts]: Fixed

File: src/selectors/tasks/select_newly_recommended_unsaved_tasks.ts [hunk 7]
```diff
@@ -3,16 +3,18 @@ import { pickSavedTaskIds } from './pick_saved_task_ids';
...
 
 export const selectNewlyRecommendedUnsavedTasks = (appStore: Store): ReadonlyArray<Task> => {
-    const tasks = pickTasks(appStore);
-    const savedTaskIds = pickSavedTaskIds(appStore);
-    const newAnswers = pickAnswers(appStore);
     const oldAnswers = pickQuestionnaire(appStore).oldAnswers;
+    const newAnswers = pickAnswers(appStore);
+    const tasks = pickTasks(appStore);
+    const newlyRecommendedTasks = getNewlyRecommendedTasks(oldAnswers, newAnswers, tasks);
+    const newlyRecommendedUnsaveTasks = rejectTaskIds(newlyRecommendedTasks, pickSavedTaskIds(appStore));
```
Comment (rasmus-storjohann-PG) [inline src/selectors/tasks/select_newly_recommended_unsaved_tasks.ts]: `rejectTasksWithIds()`

**Pull request merged**